mail/spring-cloud-starter-stream-source-mail/src/main/java/org/springframework/cloud/stream/app/mail/source/MailSourceProperties.java

```diff
@@ -0,0 +1,220 @@
+ * @author Amol
+ */
+@ConfigurationProperties
+public class MailSourceProperties {
+
+	/**
+	 * Protocol to be used for recieving mail.
+	 */
+	private String protocol;
+
+	/**
+	 * Username for email account. Usually is email id
+	 */
+	private String username;
+
+	/**
+	 * Password for email account.
+	 */
+	private String password;
+
+	/**
+	 * Host URL for email access
+	 */
+	private String host;
+
+	/**
+	 * Port to be used for email e.g. 993 for imaps
+	 */
+	private String port;
+
+	/**
+	 * Email folder to be scanned for email
+	 */
+	private String folder;
+
+	/**
+	 * Set to true to mark email as read.
+	 */
+	private boolean markAsRead = false;
+
+	/**
+	 * Set to true to delete email after download.
+	 */
+	private boolean delete = false;
+
+	/**
+	 * Comma seperated list of javaMail properties to be used.
+	 */
+	private String javaMailProperties;
+
+	/**
+	 * @return the protocol
+	 */
+	public String getProtocol() {
+		return protocol;
+	}
+
+	/**
+	 * @param protocol the protocol to set
+	 */
+	public void setProtocol(String protocol) {
+		this.protocol = protocol;
+	}
+
+	/**
+	 * @return the username
+	 */
+	public String getUsername() {
+		return username;
+	}
+
+	/**
+	 * @param username the username to set
+	 */
+	public void setUsername(String username) {
+		this.username = username;
+	}
+
+	/**
+	 * @return the password
+	 */
+	public String getPassword() {
+		return password;
+	}
+
+	/**
+	 * @param password the password to set
+	 */
+	public void setPassword(String password) {
+		this.password = password;
+	}
+
+	/**
+	 * @return the host
+	 */
+	public String getHost() {
+		return host;
+	}
+
+	/**
+	 * @param host the host to set
+	 */
+	public void setHost(String host) {
+		this.host = host;
+	}
+
+	/**
+	 * @return the port
+	 */
+	public String getPort() {
+		return port;
+	}
+
+	/**
+	 * @param port the port to set
+	 */
+	public void setPort(String port) {
+		this.port = port;
+	}
+
+	/**
+	 * @return the folder
+	 */
+	public String getFolder() {
+		return folder;
+	}
+
+	/**
+	 * @param folder the folder to set
+	 */
+	public void setFolder(String folder) {
+		this.folder = folder;
+	}
+
+	/**
+	 * @return the markAsRead
+	 */
+	public boolean isMarkAsRead() {
+		return markAsRead;
+	}
+
+	/**
+	 * @param markAsRead the markAsRead to set
+	 */
+	public void setMarkAsRead(boolean markAsRead) {
+		this.markAsRead = markAsRead;
+	}
+
+	/**
+	 * @return the delete
+	 */
+	public boolean isDelete() {
+		return delete;
+	}
+
+	/**
+	 * @param delete the delete to set
+	 */
+	public void setDelete(boolean delete) {
+		this.delete = delete;
+	}
+
+	/**
+	 * @return the javaMailProperties
+	 */
+	public String getJavaMailProperties() {
+		return javaMailProperties;
+	}
+
+	/**
+	 * @param javaMailProperties the javaMailProperties to set
+	 */
+	public void setJavaMailProperties(String javaMailProperties) {
+		this.javaMailProperties = javaMailProperties;
+	}
+
+	protected Map<String, String> parsePropertiesString() {
```

Comment:
I wonder if we could follow with existing approach in the `ScriptVariableProperties` which is applied onto the `spring-cloud-starter-stream-processor-groovy-transform` for example. I mean something like this:
```
/**
  * Variable bindings as a new line delimited string of name-value pairs, e.g. 'foo=bar\n baz=car'.
*/
private Properties variables;
```
Where we receive `Properties` as is and just inject them into the target adapter.
All other *hard* parsing logic is done during properties population by Boot.